From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 10 Jun 2019 08:29:11 +0200
Subject: Revert "doc: Cope with the deprecation of AutoReporter"
Patch-mainline: never, 5.1.8 fix
References: fix kernel-doc

This reverts commit 32a4c9e04c09af21d423ab57a5819b854fa5f3e2, upstream
commit 2404dad1f67f8917e30fc22a85e0dbcc85b99955. It breaks kernel-docs
build with 5.1.x (and not with 5.2):
Documentation/gpu/i915.rst:403: WARNING: Title level inconsistent:

Global GTT Fence Handling
~~~~~~~~~~~~~~~~~~~~~~~~~

reST markup error:
Documentation/gpu/i915.rst:403: (SEVERE/4) Title level inconsistent:

Global GTT Fence Handling
~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Documentation/sphinx/kerneldoc.py | 34 ++++++++-----------------------
 1 file changed, 8 insertions(+), 26 deletions(-)

diff --git a/Documentation/sphinx/kerneldoc.py b/Documentation/sphinx/kerneldoc.py
index 1159405cb920..200c8aa4a04f 100644
--- a/Documentation/sphinx/kerneldoc.py
+++ b/Documentation/sphinx/kerneldoc.py
@@ -37,17 +37,7 @@ import glob
 from docutils import nodes, statemachine
 from docutils.statemachine import ViewList
 from docutils.parsers.rst import directives, Directive
-
-#
-# AutodocReporter is only good up to Sphinx 1.7
-#
-import sphinx
-
-Use_SSI = sphinx.__version__[:3] >= '1.7'
-if Use_SSI:
-    from sphinx.util.docutils import switch_source_input
-else:
-    from sphinx.ext.autodoc import AutodocReporter
+from sphinx.ext.autodoc import AutodocReporter
 
 import kernellog
 
@@ -135,7 +125,13 @@ class KernelDocDirective(Directive):
                     lineoffset += 1
 
             node = nodes.section()
-            self.do_parse(result, node)
+            buf = self.state.memo.title_styles, self.state.memo.section_level, self.state.memo.reporter
+            self.state.memo.reporter = AutodocReporter(result, self.state.memo.reporter)
+            self.state.memo.title_styles, self.state.memo.section_level = [], 0
+            try:
+                self.state.nested_parse(result, 0, node, match_titles=1)
+            finally:
+                self.state.memo.title_styles, self.state.memo.section_level, self.state.memo.reporter = buf
 
             return node.children
 
@@ -144,20 +140,6 @@ class KernelDocDirective(Directive):
                            (" ".join(cmd), str(e)))
             return [nodes.error(None, nodes.paragraph(text = "kernel-doc missing"))]
 
-    def do_parse(self, result, node):
-        if Use_SSI:
-            with switch_source_input(self.state, result):
-                self.state.nested_parse(result, 0, node, match_titles=1)
-        else:
-            save = self.state.memo.title_styles, self.state.memo.section_level, self.state.memo.reporter
-            self.state.memo.reporter = AutodocReporter(result, self.state.memo.reporter)
-            self.state.memo.title_styles, self.state.memo.section_level = [], 0
-            try:
-                self.state.nested_parse(result, 0, node, match_titles=1)
-            finally:
-                self.state.memo.title_styles, self.state.memo.section_level, self.state.memo.reporter = save
-
-
 def setup(app):
     app.add_config_value('kerneldoc_bin', None, 'env')
     app.add_config_value('kerneldoc_srctree', None, 'env')
-- 
2.21.0

